### PR TITLE
Feat/keysets

### PR DESCRIPTION
**Pull request by emmanueldenloye**
This is just the first set of commits from me. I've added the basic definition for keysets and also added a sorely needed .gitignore file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+TAGS
+dist-newstyle/*
+dist/*
```
Comment (emilypi) [inline .gitignore]: could you add this to your `gitignore_global` instead?

File: src/Pact/AST/Literals.hs [hunk 2]
```diff
@@ -32,14 +32,29 @@ import GHC.Generics
...
+newtype KeySetPredicate = KeysetPredicate
+  { getPredicate :: Text
+  } deriving (Eq, Ord, Show, Generic, NFData)
+
+newtype PublicKey = PublicKey
+  { getPublicKey :: ByteString
+  } deriving (Eq, Ord, Show, Generic, NFData)
+
+data KeySet = KeySet
+  { ksPublicKeys :: !(NESet PublicKey)
```
Comment (emilypi) [inline src/Pact/AST/Literals.hs]: This is not something i'd care to make strict, simply because the set can be rather large.

**Review by emilypi — APPROVED** [review]

File: src/Pact/AST/Literals.hs [hunk 3]
```diff
@@ -32,14 +32,29 @@ import GHC.Generics
...
+  , ksPredicate :: {-# UNPACK #-}!KeySetPredicate
+  } deriving (Eq, Ord, Show, Generic, NFData)
+
```
Comment (emilypi) [inline src/Pact/AST/Literals.hs:46]: On second thought, lets add this to its own file `KeySet.hs`

File: .gitignore (hunk 1 above)
Comment (emmanueldenloye) [reply .gitignore]: Ok, that works.

File: src/Pact/AST/Literals.hs (hunk 2 above)
Comment (emmanueldenloye) [reply src/Pact/AST/Literals.hs]: That's a good point.

**Pull request merged**